prov/gni/src/gnix_av.c

```diff
@@ -285,7 +300,7 @@ static int map_insert(struct gnix_fid_av *int_av, const void *addr,
 }
 
 /*
- * TODO: Actually implement once the address is being stored.
+ * TODO: slab should be freed when entries in the slab drop to zero
  */
```

Comment:
Since you're adding TODO's, you could also keep around the empty slabs for future insertions.